apps/src/templates/teacherDashboard/SectionScriptProgress.js

```diff
@@ -0,0 +1,82 @@
+import React, { PropTypes, Component } from 'react';
+import ProgressBubbleSet from '@cdo/apps/templates/progress/ProgressBubbleSet';
+import { levelsByLesson } from '@cdo/apps/code-studio/progressRedux';
+
+const styles = {
+  bubbles: {
+    overflowX: 'scroll',
+    whiteSpace: 'nowrap',
+  }
+};
+
+/**
+ * Displays a section's progress for a particular script once we have data
+ */
+export default class SectionScriptProgress extends Component {
+  static propTypes = {
+    section: PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      students: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        name: PropTypes.string.isRequired,
+      })).isRequired
```

Comment:
it's possible this array is empty - will that still satisfy isRequired?

Reply:
I think so.